packages/flutter/lib/src/widgets/_window.dart

```diff
@@ -620,6 +621,238 @@ abstract class DialogWindowController extends BaseWindowController {
   void setMinimized(bool minimized);
 }
 
+/// Delegate class for tooltip window controller.
+///
+/// {@macro flutter.widgets.windowing.experimental}
+///
+/// See also:
+///
+/// * [TooltipWindowController], the controller that creates and manages tooltip windows.
+/// * [TooltipWindow], the widget for a tooltip window.
+/// * [RegularWindowControllerDelegate], the delegate for regular window controllers.
+mixin class TooltipWindowControllerDelegate {
+  /// Invoked when the user attempts to close the window.
+  ///
+  /// The default implementation destroys the window. Subclasses
+  /// can override the behavior to delay or prevent the window from closing.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  ///
+  /// See also:
+  ///
+  /// * [onWindowDestroyed], which is invoked after the window is closed.
+  @internal
+  void onWindowCloseRequested(TooltipWindowController controller) {
+    if (!isWindowingEnabled) {
+      throw UnsupportedError(_kWindowingDisabledErrorMessage);
+    }
+
+    controller.destroy();
+  }
+
+  /// Invoked after the window is closed.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  ///
+  /// See also:
+  ///
+  /// * [onWindowCloseRequested], which is invoked when the user attempts to close the window.
+  @internal
+  void onWindowDestroyed() {
+    if (!isWindowingEnabled) {
+      throw UnsupportedError(_kWindowingDisabledErrorMessage);
+    }
+  }
+}
+
+/// A controller for a tooltip window.
+///
+/// A tooltip window is a small window that displays brief, informative text
+/// when a user hovers over or focuses on a UI element. Tooltip windows are
+/// typically used to provide additional context or explanations for UI elements
+/// without cluttering the main interface. As such, the a may not receive input
+/// focus from the user. It will however stay open when another window receives
+/// input focus.
+///
+/// This class does not interact with the widget tree. Instead, it is typically
+/// provided to the [TooltipWindow] widget, which renders the content inside the
+/// tooltip window.
+///
+/// The user of this class is responsible for managing the lifecycle of the window.
+/// When the window is no longer needed, the user should call [destroy] on this
+/// controller to release the resources associated with the window.
+///
+/// {@tool snippet}
+/// An example usage might look like:
+///
+/// ```dart
+/// // TODO(mattkae): remove invalid_use_of_internal_member ignore comment when this API is stable.
+/// // ignore_for_file: invalid_use_of_internal_member
+/// import 'package:flutter/material.dart';
+/// import 'package:flutter/src/widgets/_window.dart';
+/// import 'package:flutter/src/widgets/_window_positioner.dart';
```

Comment:
We'll want to clean this sample up before we make these APIs public. However, I suspect we'll likely forget this.

What's the best way to remember? What do you think of filing an issue that's a checklist of things we need to do before we make windowing APIs public, and then linking to that issue here?